### DISCUSSION TITLE
Example to use internal temperature sensor of stm32h750b_dk

### DISCUSSION THREAD

**Discussion by MaazSk** [zephyrproject-rtos/zephyr#77780] [Q&A]
Hello Everyone,

I found there is some code regarding stm32_temperature sensor [over here.](https://github.com/zephyrproject-rtos/zephyr/blob/v3.7.0/drivers/sensor/st/stm32_temp/stm32_temp.c) Also, there is an test for temperature sensor under tests/sensors/temp_sensor but this test does not work directly with stm32h750b_dk I tried using nucleo_u575zi also but it did not work.

So, wanted to know if there is any direct support to use internal temperature sensor or what should be the method to use it

**Comment by FRASTM** [comment]
Hi @MaazSk 

We usually use the samples/sensor/die_temp_polling/   to  measure the die temperature with the ADC internal sensor channel
For the stm32h750b_dk, you need to specify the DTS with an overlay file like this ./samples/sensor/die_temp_polling/ boards/stm32h750b_dk.overlay

```
/ {
	aliases {
		die-temp0 = &die_temp;
	};
};

&die_temp {
	status = "okay";
};

&adc3 {
	pinctrl-0 = <&adc3_inn3_pf8>;
	pinctrl-names = "default";

	st,adc-clock-source = <SYNC>;
	st,adc-prescaler = <4>;
	status = "okay";
};
```

**Reply by MaazSk** [reply]
Testing the same in a moment thank you!

**Reply by MaazSk** [reply]
I have a small doubt, I already have an ADC initialized to ADC3 and if I don't use `adc3_inn3_pf8` still I get the temperature readings. So, what is the purpose of `adc3_inn3_pf8`?

Also, while I am reading internal ADC channels why is `pinctrl-0 = <&adc3_inn3_pf8>;` required?

**Comment by FRASTM** [comment]
with that overlay, west build  -p auto -b stm32h750b_dk samples/sensor/die_temp_polling ; west flash 

```
*** Booting Zephyr OS build v3.7.0-1880-g25e90e7bb0ef ***
CPU Die temperature[dietemp]: 32.1 °C
```

**Comment by FRASTM** [comment]
Same method for the nucleo_u575 board where the die-temp is on adc1 channel 19. A  nucleo_u575zi_q.overlay    file  like the following should be enough : 

```
/ {
	aliases {
		die-temp0 = &die_temp;
	};
};

&die_temp {
	status = "okay";
};
```

**Comment by FRASTM** [comment]
See PR  https://github.com/zephyrproject-rtos/zephyr/pull/77619
